src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -281,18 +293,18 @@ def _callback(self, header, err):
             logger.error("Could not sent: %s, %s", header, err)
 
     def _parse_file_name(self, path):
-        matched = parse_file_path(self.file_regex, path)
         try:
-            data_source = get_entry(matched, "data_source")
-        except utils.UsageError:
+            matched = parse_file_path(self.file_regex, path)
             if self.default_data_source is not None:
                 data_source = self.default_data_source
             else:
-                raise
+                data_source = get_entry(matched, "data_source")
```

Comment:
Now `data_source` is always ignored if `default_data_source` is set. It should be the other way around.